numbast/src/numbast/static/struct.py

```diff
@@ -103,6 +103,17 @@ def ctor_impl(context, builder, sig, args):
     return builder.load(selfptr, align=getattr({struct_type_name}, "alignof_", None))
     """
 
+    struct_conversion_ctor_lowering_template = """
+@lower_cast({param_types}, {struct_type_name})
+def conversion_impl(context, builder, fromty, toty, value):
+    return ctor_impl(
```

Comment:
Unfortunately, I think `ctor_impl` is implemented in a local function scope. For example:

```
def lowerXXX():
    def shim_function: ...
    def shim_function_caller: ...
    def ctor_impl: ...
````

So you may need to add this code into the same scope, see `lowering_body_template` below.

Reply:
Not needed because this lower cast section is conditionally appended to the lower body code section only when there is an implicit ctor.